src/IntervalTrees.jl

```diff
@@ -1038,22 +1042,22 @@ function Base.findfirst(t::LeafNode{K, V, B}, key::AbstractInterval{K}, f) where
           first(t.keys[i]) == first(key) &&
           last(t.keys[i]) == last(key)
         if f(t.entries[i], key)
-            return Nullable(t.entries[i])
+            return t.entries[i]
         end
         i += 1
-        if i > length(t) && !isnull(t.right)
-            t = get(t.right)
+        if i > length(t) && t.right !== nothing
+            t = t.right
             i = 1
         end
     end
 
-    return Nullable{V}()
+    return nothing
 end
 
 
 function Base.findfirst(t::InternalNode{K, V, B}, key::AbstractInterval{K}, f) where {K, V, B}
     i = findidx(t, key)
-    if i <= length(t) - 1 && key >= t.keys[i]
+    if i <= length(t) - 1 && !(key < t.keys[i])
```

Comment:
Please review this code in light of #36. While this is just a workaround a complete elaborate fix will needed as described in #36.